homeassistant/components/sensor/filter.py

```diff
@@ -137,7 +148,20 @@ def filter_sensor_state_listener(entity, old_state, new_state):
                 self._unit_of_measurement = new_state.attributes.get(
                     ATTR_UNIT_OF_MEASUREMENT)
 
-            self.async_schedule_update_ha_state()
+            if update_ha:
+                self.async_schedule_update_ha_state()
+
+        if self._history_period:
+            start = dt_util.as_utc(datetime.now() - self._history_period)
+
+            history_list = history.state_changes_during_period(
```

Comment:
You're inside an async function and are calling a sync function: move this to the executor:

```python
from functools import partial

history_list = await self.hass.async_add_job(partial(history.state_changes_during_period, self.hass, start, entity_id=self._entity))
```

(you have to use partial because of the keyword parameter)